test/proxyquire-argumentvalidation.js

```diff
@@ -63,4 +63,18 @@ describe('Illegal parameters to resolve give meaningful errors', function () {
       throws(act, /Invalid stub: "myStub" cannot be undefined/)
     })
   })
+
+  describe('when I pass a stub with a key that is not required on the proxyquired object', function () {
+    function act () {
+      const proxyquireThrowOnUnresolved = proxyquire.throwOnUnresolved()
```

Comment:
```suggestion
      proxyquire.throwOnUnresolved()
```

The return value is the main fn itself—assigning the return value implies a new copy of proxyquire which isn't the case here. This test mutates state it doesn't own and I'd rather that be apparent in case it causes future issues.

Reply:
Ah yes, thanks for pointing that out! I've been using a generator function most of the time around proxyquire to have a "clean slate" for each test run so forgot it was actually mutating here!